features/region-segmentation-comparison-plugin/src/polus/plugins/features/region_segmentation_comparison/evaluate.py

```diff
@@ -0,0 +1,719 @@
+        FP = [0] * (input_classes + 1)
+        FN = [0] * (input_classes + 1)
+        total_over_segmented = [0] * (input_classes + 1)
+        total_under_segmented = [0] * (input_classes + 1)
+
+    if individual_summary:
+        ind_sum = []
+
+    if total_summary:
+        total_iou = [0] * (input_classes + 1)
+        total_tpr = [0] * (input_classes + 1)
+        total_precision = [0] * (input_classes + 1)
+        total_fnr = [0] * (input_classes + 1)
+        total_fdr = [0] * (input_classes + 1)
+        total_fscore = [0] * (input_classes + 1)
+        total_f1_score = [0] * (input_classes + 1)
+        total_fmi = [0] * (input_classes + 1)
+
+    try:
+        for file in fp():
+            file_name = file[1][0]
+            tile_grid_size = 1
+            tile_size = tile_grid_size * 2048
+            with BioReader(file_name, max_workers=cpu_count()) as br_pred:
+                with BioReader(
+                    pathlib.Path(gt_dir / file_name.name),
+                    max_workers=cpu_count(),
+                ) as br_gt:
+                    logger.info(f"Evaluating image {file_name}")
+                    total_files += 1
+
+                    if individual_summary:
+                        mean_centroid = [0] * (input_classes + 1)
+                        mean_iou = [0] * (input_classes + 1)
+
+                    totalCells = [0] * (input_classes + 1)
+                    tp = [0] * (input_classes + 1)
+                    fp = [0] * (input_classes + 1)
+                    fn = [0] * (input_classes + 1)
+                    over_segmented = [0] * (input_classes + 1)
+                    under_segmented = [0] * (input_classes + 1)
+                    for z in range(br_gt.Z):
+                        # Loop across the length of the image
+                        for y in range(0, br_gt.Y, tile_size):
+                            y_max = min([br_gt.Y, y + tile_size])
+                            for x in range(0, br_gt.X, tile_size):  # noqa
+                                x_max = min([br_gt.X, x + tile_size])
+                                im_gt = np.squeeze(
+                                    br_gt[y:y_max, x:x_max, z : z + 1, 0, 0]  # noqa
+                                )
+                                im_pred = np.squeeze(
+                                    br_pred[y:y_max, x:x_max, z : z + 1, 0, 0]  # noqa
+                                )
+
+                                if input_classes > 1:
+                                    classes = np.unique(im_gt)
+                                else:
+                                    classes = [1]
+                                for cl in classes:
+                                    if len(classes) > 1:
+                                        im_pred = np.where(im_pred == cl, cl, 0)
+                                        im_gt = np.where(im_gt == cl, cl, 0)
+                                        im_pred, _, _, _ = ccl(np.uint8(im_pred))
```

Comment:
np.uint8() as a function results in mypy type errors. It is recommended to use:
`impred.astype(np.uint8)`